addons/website_sale/models/product.py

```diff
@@ -200,7 +200,7 @@ def _website_price(self):
         ret = self.env.user.has_group('sale.group_show_price_subtotal') and 'total_excluded' or 'total_included'
 
         for p, p2 in zip(self, self2):
-            taxes = partner.property_account_position_id.map_tax(p.taxes_id).filtered(lambda x: x.company_id == company_id)
+            taxes = partner.property_account_position_id.map_tax(p.sudo().taxes_id.filtered(lambda x: x.company_id == company_id))
```

Comment:
@pedrobaeza I'm commiting the patch without `sudo()` so you can better check behaviour. Also summarizing the behavior this are the steps to reproduce (company1 and company2 for the example):

- If no `sudo()` is applied:

1.  403 error trying to access to the shop due to permission of public user.

- Before patch:

1. Enable b2c prices
2. Create tax1 linked to company1
3. Create tax2 linked to company2
4. From company1 add tax1 to productA
5. From company2 add tax2 to productA
6. If a user is present in both companies (either public or not): price = list_price + tax1 + tax2

- After patch:

1. Same first steps.
2. If a user is present in both companies (either public or not): price = list_price + tax(belonging to website company)